urllib3/poolmanager.py

```diff
@@ -186,9 +199,9 @@ def connection_from_context(self, request_context):
         pool_key_constructor = self.key_fn_by_scheme[scheme]
         pool_key = pool_key_constructor(request_context)
 
-        return self.connection_from_pool_key(pool_key)
+        return self.connection_from_pool_key(pool_key, request_context=request_context)
```

Comment:
The pool key is a subset of values from the context at the moment. The HTTP pool key is on the following keywords from the ``request_context`` dictionary:

* scheme
* host
* port
* timeout
* retries
* strict
* block
* source_address

However, in addition to those fields the context may also have one or more key-value pairs meant for the ``HTTPConnectionPool`` class which accepts all those keyword arguments (except for scheme) and:

* maxsize
* headers
* _proxy
* _proxy_header

On top of that, any additional keywords provided to ``HTTPConnectionPool`` get handed on to ``HTTPConnection``, which accepts ``(*args, **kwargs)``, but looks for:

* source_address
* socket_options

It has just occurred to me that since we don't key on ``maxsize``, for example, I think changing this after a pool has been created won't cause a new pool with a different ``maxsize`` to be created. It's not something I would have considered doing, but I guess that would be surprising to someone. Maybe we _should_ key on the entire context.

What might make this all cleaner is to have an object that contains what ``context`` currently holds and also knows how to derive a pool key from it so that we aren't handing both the pool key and the context around.